bytecomp/translclass.ml

```diff
@@ -430,37 +428,39 @@ let rec transl_class_rebind obj_init cl vf =
                    body = Matching.for_function
                             pat.pat_loc None (Lvar param) [pat, rem] partial}
       in
-      (path,
+      (path, path_lam,
        match obj_init with
          Lfunction {kind = Curried; params; body} -> build params body
        | rem                                      -> build [] rem)
   | Tcl_apply (cl, oexprs) ->
-      let path, obj_init = transl_class_rebind obj_init cl vf in
-      (path, transl_apply obj_init oexprs Location.none)
+      let path, path_lam, obj_init = transl_class_rebind obj_init cl vf in
+      (path, path_lam, transl_apply obj_init oexprs Location.none)
   | Tcl_let (rec_flag, defs, _vals, cl) ->
-      let path, obj_init = transl_class_rebind obj_init cl vf in
-      (path, Translcore.transl_let rec_flag defs obj_init)
+      let path, path_lam, obj_init = transl_class_rebind obj_init cl vf in
+      (path, path_lam, Translcore.transl_let rec_flag defs obj_init)
   | Tcl_structure _ -> raise Exit
   | Tcl_constraint (cl', _, _, _, _) ->
-      let path, obj_init = transl_class_rebind obj_init cl' vf in
+      let path, path_lam, obj_init = transl_class_rebind obj_init cl' vf in
       let rec check_constraint = function
           Cty_constr(path', _, _) when Path.same path path' -> ()
         | Cty_arrow (_, _, cty) -> check_constraint cty
         | _ -> raise Exit
       in
       check_constraint cl.cl_type;
-      (path, obj_init)
+      (path, path_lam, obj_init)
   | Tcl_open (_, _, _, _, cl) ->
       transl_class_rebind obj_init cl vf
 
 let rec transl_class_rebind_0 self obj_init cl vf =
   match cl.cl_desc with
     Tcl_let (rec_flag, defs, _vals, cl) ->
-      let path, obj_init = transl_class_rebind_0 self obj_init cl vf in
-      (path, Translcore.transl_let rec_flag defs obj_init)
+      let path, path_lam, obj_init =
+        transl_class_rebind_0 self obj_init cl vf
+      in
+      (path, path_lam, Translcore.transl_let rec_flag defs obj_init)
   | _ ->
-      let path, obj_init = transl_class_rebind obj_init cl vf in
-      (path, lfunction [self] obj_init)
+      let path, path_lam, obj_init = transl_class_rebind obj_init cl vf in
+      (path, path_lam, lfunction [self] obj_init)
```

Comment:
Another minor remark about something that I found a bit confusing at first:
- the result of `transl` functions has the form `(path, path_lam, obj_init)` where the last two values are lambda expressions (the first to access the path at runtime, the second one is the function that initializes an object)
- the accumulator of inherited initialisations is a list where elements have the form `(obj_init, path, path_lam)`, with the same meaning as above.
Reading closely I don't think that any mistake is caused by this permutation.

Reply:
Fixed